### ISSUE TITLE
 Error when compiling project with NET SDK 9.0.100: No file exists for the asset at either location

### ISSUE THREAD

**Issue by pacoweb** [failwyn/BundlerMinifier#41]
### Installed product versions
Microsoft Visual Studio Enterprise 2022 (64-bit) - Current
Version 17.12.0
**NET SDK 9.0.100**
 BuildBundlerMinifier2022, Version 2.9.11

### Description
The new version of Visual Studio 2022 17.12.0 has removed the .NET SDK 8 and installed the **NET SDK 9.0.100**
After that I get an error when compiling the project
_System.InvalidOperationException: No file exists for the asset at either location 'wwwroot\css\site.min.css' or 'wwwroot\css\site.min.css'.
   at Microsoft.AspNetCore.StaticWebAssets.Tasks.StaticWebAsset.ComputeFingerprintAndIntegrity(String identity, String originalItemSpec)
   at Microsoft.AspNetCore.StaticWebAssets.Tasks.DefineStaticWebAssets.Execute()	WebApplication3	C:\Program Files\dotnet\sdk\9.0.100\Sdks\Microsoft.NET.Sdk.StaticWebAssets\targets\Microsoft.NET.Sdk.StaticWebAssets.targets	662_	

I have created an example project with only a static file to compress: site.css
This is the bundleconfig.json:
```
[
    {
        "outputFileName": "wwwroot/css/site.min.css",
        "inputFiles": [
            "wwwroot/css/site.css"
        ]
    }
]

```
The extension works correctly: https://marketplace.visualstudio.com/items?itemName=Failwyn.BundlerMinifier64

I attach a project to reproduce the error
[WebErrorExample.zip](https://github.com/user-attachments/files/17745131/WebErrorExample.zip)

**Comment by dotnetshadow** [comment]
I have experienced the same issue I had to rollback my visual studio and everything worked ok again

**Comment by timotyturini** [comment]
Just adding informations: for some reason -and only on net9.0 (<=net8.0 is working)-, the gzip output generates a separate, different type of error; anyway looks like an issue related to the SDK rather than this library package, because if i uninstall the BuildBundlerMinifier2022 and keep all pre-generated bundles -including their gzipped version- the compilation fails every time.

**Comment by dotnetshadow** [comment]
@timotyturini Is there any fix for this? Or have you used another package to get around the issue?

**Comment by timotyturini** [comment]
Hi @dotnetshadow 
Sadly i don't think there is an alternatives atm.

Visual Studio 17.12 itself introduced half the problem, as it refuses to **recompile** a solution with any BuildBundlerMinifier2022 installed, but surprisingly working by **clean** solution and **compile**.

The other half of the problem is NET9.0 SDK refusing to handle .gz files in the project.

My current solution is to stick to NET8.0 SDK as long as possible and cringe everytime i click **recompile** by accident

**Comment by pacoweb** [comment]
@dotnetshadow,  @timotyturini,, @failwyn 
Hi, to fix this just add this code in csproj

`<PropertyGroup>
<PrepareForBuildDependsOn>
BundleMinify;
</PrepareForBuildDependsOn>
</PropertyGroup>`

Tested with the example solution that I put when opening the incident

Thanks to: 
https://developercommunity.visualstudio.com/t/InvalidOperationException-REBUILD-fails-/10799774#T-N10802978

**Comment by timotyturini** [comment]
That worked in VS 17.12.2, with a NET8.0 project and even a NET9.0 one (without .gz bundles)

Thank you so much @pacoweb 

![image](https://github.com/user-attachments/assets/9a9f14ee-7935-42f8-b38a-61a1b23d8fd5)

The last thing missing at this point is the NET9.0 SDK allowing us to have .gz "nested" bundle files in the project.
The error in this case is an InvalidOperationException
caused by SingleOrDefault()
in Microsoft.AspNetCore.StaticWebAssets.Tasks.GenerateStaticWebAssetsDevelopmentManifest.<ComputeManifestAssets>d_19.MoveNext()

By the way, this issue is not strictly related to this thread.
I think this can be closed now.

**Comment by failwyn** [comment]
Reopening this issue to make it easier to find the workaround

Add the following to your project file to fix this until a proper fix is found.

`<PropertyGroup> <PrepareForBuildDependsOn> WebCompile;$(PrepareForBuildDependsOn) </PrepareForBuildDependsOn> </PropertyGroup>`

Build => Clean Solution
Build => Rebuild Solution

https://github.com/dotnet/aspnetcore/issues/58948

**Comment by wgutierrezr** [comment]
I am having the same issue VS2022 17.13.6

**Comment by failwyn** [comment]
@wgutierrezr Does it work if you Clean Solution, then Rebuild Solution?

**Comment by wgutierrezr** [comment]
No, it does not. I am using Net 8, it works only when a switch back to Net 6

**Comment by failwyn** [comment]
@wgutierrezr  Is there any way you can provide a sample project to replicate this? The Clean & Rebuild works for most of the people on this issue, including myself.

**Comment by wgutierrezr** [comment]
It is a huge project sorry.

**Comment by failwyn** [comment]
@wgutierrezr no worries, can you provide the bundleconfig.json?

**Comment by wgutierrezr** [comment]
[
  {
    "outputFileName": "wwwroot/css/thirdparty.bundle.css",
    "inputFiles": [
      "node_modules/jquery-ui-dist/jquery-ui.min.css",
      "node_modules/devextreme/dist/css/dx.common.css",
      "node_modules/devextreme/dist/css/dx.light.css"
    ],
    "minify": {
      "enabled": false,
      "adjustRelativePaths": false
    }
  },
  {
    "outputFileName": "wwwroot/css/viewer.part.bundle.css",
    "inputFiles": [
      "node_modules/@devexpress/analytics-core/dist/css/dx-analytics.common.css",
      "node_modules/@devexpress/analytics-core/dist/css/dx-analytics.light.css",
      "node_modules/devexpress-reporting/dist/css/dx-webdocumentviewer.css"
    ],
    "minify": {
      "enabled": false,
      "adjustRelativePaths": false
    }
  },

  {
    "outputFileName": "wwwroot/js/thirdparty.bundle.js",
    "inputFiles": [
      "node_modules/jquery/dist/jquery.min.js",
      "node_modules/jquery-ui-dist/jquery-ui.min.js",
      "node_modules/knockout/build/output/knockout-latest.js",
      "node_modules/cldrjs/dist/cldr.js",
      "node_modules/cldrjs/dist/cldr/event.js",
      "node_modules/cldrjs/dist/cldr/supplemental.js",
      "node_modules/cldrjs/dist/cldr/unresolved.js",
      "node_modules/globalize/dist/globalize.js",
      "node_modules/globalize/dist/globalize/message.js",
      "node_modules/globalize/dist/globalize/number.js",
      "node_modules/globalize/dist/globalize/currency.js",
      "node_modules/globalize/dist/globalize/date.js",
      "node_modules/uuid/dist/umd/uuidv4.min.js",
      "node_modules/exceljs/dist/exceljs.min.js",
      "node_modules/file-saver/dist/FileSaver.min.js",
      "node_modules/@babel/polyfill/dist/polyfill.min.js",
      "node_modules/jszip/dist/jszip.js",
      "node_modules/devextreme/dist/js/dx.all.js",
      "node_modules/devextreme/dist/js/aspnet/dx.aspnet.data.js",
      "node_modules/devextreme/dist/js/aspnet/dx.aspnet.mvc.js",


      "node_modules/ace-builds/src-min-noconflict/ace.js",
      "node_modules/ace-builds/src-min-noconflict/ext-language_tools.js",
      "node_modules/ace-builds/src-min-noconflict/theme-dreamweaver.js",
      "node_modules/ace-builds/src-min-noconflict/theme-ambiance.js",
      "node_modules/ace-builds/src-min-noconflict/snippets/text.js"
    ],
    "minify": {
      "enabled": false
    },
    "sourceMap": false
  },
  {
    "outputFileName": "wwwroot/js/viewer.part.bundle.js",
    "inputFiles": [
      "node_modules/@devexpress/analytics-core/dist/js/dx-analytics-core.min.js",
      "node_modules/devexpress-reporting/dist/js/dx-webdocumentviewer.min.js"
    ],
    "minify": {
      "enabled": false
    },
    "sourceMap": false
  }
]
